### PR TITLE
Add grpc-swift

### PR DESCRIPTION
**Pull request by glbrntt**
### Pull Request Description

Add grpc-swift

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 4.0 and passes any unit tests
  - 5.2+
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
  - Apache License, version 2.0
- [x] pass `./project_precommit_check` script run

```
========================================
Action Summary:
     Passed: 1
     Failed: 0
    XFailed: 0
    UPassed: 0
      Total: 1
========================================
Repository Summary:
      Total: 1
========================================
Result: PASS
========================================
--- grpc-swift checked successfully against Swift 5.2 ---
```

### CODE REVIEW

**Review by clackary — APPROVED** [review]

**Comment by clackary** [comment]
@swift-ci please test

**Review by ktoso — APPROVED** [review]

**Comment by glbrntt** [comment]
Which toolchain is the CI run with, I couldn't find anything in the logs?

I suspect the grpc-swift build timed out because of a bug in 5.3.1, it compiles fine on 5.3.2.

**Comment by xedin** [comment]
Sout compat suite builds a toolchain from the HEAD of main branch and uses it to build the projects.

**Comment by glbrntt** [comment]
Bug filed for the compilation failure: https://bugs.swift.org/browse/SR-14068

File: projects.json [hunk 1]
```diff
@@ -834,6 +834,37 @@
...
+    ],
+    "platforms": [
+      "Darwin",
+      "Linux"
+    ],
+    "actions": [
+      {
+        "action": "BuildSwiftPackage",
+        "configuration": "release",
+        "tags": "sourcekit-disabled swiftpm"
```
Comment (clackary) [inline projects.json:861]:
```suggestion
        "tags": "sourcekit-disabled swiftpm",
        "xfail": {
            "issue": "https://bugs.swift.org/browse/SR-14068",
            "compatibility": ["5.2", "5.3"],
            "branch": ["main"]
        }
```

Comment (clackary) [reply projects.json:861]: We can still merge this if we add an xfail here since we have a bug tracking the regression.

**Comment by shahmishal** [comment]
@swift-ci test

File: projects.json [hunk 2]
```diff
@@ -834,6 +834,42 @@
...
+    "actions": [
+      {
+        "action": "BuildSwiftPackage",
+        "configuration": "release",
+        "tags": "sourcekit-disabled swiftpm",
+        "xfail": {
+            "issue": "https://bugs.swift.org/browse/SR-14068",
+            "compatibility": ["5.2", "5.3"],
+            "branch": ["main"]
+        }
```
Comment (glbrntt) [inline projects.json:866]:
```suggestion
        "tags": "sourcekit-disabled swiftpm"
```

**Comment by shahmishal** [comment]
@swift-ci test

**Pull request merged**